### PR TITLE
[PORT] [WIP-NEEDS-TESTING] [AAAAA] Overmap Part 1

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

Okay, so before the other maints come to bust down my door, I have very much tested this, though I am sad to say that the maps from horizon do not work at all here, so we're just sat with jury-rigged shitbox shuttles until I or another mapper comes along and makes some proper shuttles. (Maybe even a ship map?)

[a2d4163](https://github.com/Artea-Station/Artea-Station-Server/pull/56/commits/a2d416304a999f8a10847e4bfe5000667a633ed4) is the current "stable" commit for those who want to try this locally.

## How Does This Help ***Gameplay***?

WE CAN FUCKING GET RID OF THE MAGIC BULLSHIT LAVALAND SHIT WITHOUT REMOVING TONS OF GAMEPLAY.

Still kinda annihilates it for now, cause we inherit 0 new ruins or maps, but I hope to fix that soon, once I've ported the rest of overmap.

Oh, also, this PR is done, though the overall featureset for this is only like 1/3rd done, if that.

Oh look, more than just lavaland as mining places, oh look, there's the main system we need for hardspace and physical trading shuttles when I get around to that.

## How Does This Help ***Roleplay***?

Oh look, shuttle trips actually can be seen now, isn't that nice? Also something something downtime RP.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary> 

Okay, I don't think any amount of "proof of testing" screencaps and videos will properly document every moving part.
image
image
image
image
image
image
Unfortunately, SS13 lighting is complete shitcode, and doesn't seem to properly support non-statically lit turfs on planets.
image

</details>

## Changelog

:cl: Azarak (original), RimiNosha (port)
add: Overmap. Yea, go fly shuttles and stuff. You'll like it.
add: Dynamic weather system. Weather no longer effects every Z at once, and has support for per Z weather. plus new weather effects and sounds.
del: Old shuttle navigation.
add: A bunch of new planet types.
code: Atmos init has been overhauled, mappers beware. There should only be PLANET_ATMOS for non-station or non-special turfs from now on.
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
There's a couple of things I need to fix, but otherwise, this is in a perfectly functional state, minus actually having any overmap specific content in pre-overmap maps.

**Review by nevimer — APPROVED** [review]
looks good, we should merge this asap, but this review does not need addressing

File: code/datums/atmosphere/_atmosphere.dm [hunk 1]
```diff
@@ -50,12 +49,17 @@
...
+		if(!gaslist[gastype])
+			gaslist[gastype] = list()
+			gaslist[gastype][MOLES] = 0
 		gaslist[gastype][MOLES] += amount
 
+	/* CURRENTLY DISABLED AS I DONT KNOW WHY THIS RUNTIMES AND BREAKS A LOT OF THINGS (cannot read list)
 	// That last one put us over the limit, remove some of it
 	while(gasmix.return_pressure() > target_pressure)
-		gaslist[gastype][MOLES] -= gaslist[gastype][MOLES] * 0.1
+		gaslist[gastype][MOLES] -= 1
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/datums/atmosphere/_atmosphere.dm:60]: Yea, I think I accidentally merged a regression here.

File: code/datums/ore_node.dm [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+		if(10 to 70)
+			. = METAL_DENSITY_LOW
+		if(70 to 150)
+			. = METAL_DENSITY_MEDIUM
+		if(150 to INFINITY)
+			. = METAL_DENSITY_HIGH
+
+/datum/ore_node/proc/AttractFauna()
+	has_attracted_fauna = TRUE
+	//TODO: do this
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/datums/ore_node.dm:115]: Yea, horizon code. I've an idea on how to do it for us.

File: code/datums/transit_instance.dm [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		name_to_apply = "stranded human"
+	else if (istype(strander, /obj/structure/closet))
+		commit_strand = TRUE
+		name_to_apply = "stranded cargo"
+	else if(ismob(strander))
+		var/mob/strander_mob = strander
+		if(strander_mob.client)
+			commit_strand = TRUE
+			name_to_apply = "stranded creature"
+	if(!commit_strand)
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/datums/transit_instance.dm:134]: Nope, cause it has an ishuman and ismob check

File: code/modules/admin/verbs/possess.dm [hunk 4]
```diff
@@ -23,7 +23,8 @@
 	usr.name = O.name
 	usr.reset_perspective(O)
 	usr.control_object = O
-	O.AddElement(/datum/element/weather_listener, /datum/weather/ash_storm, ZTRAIT_ASHSTORM, GLOB.ash_storm_sounds)
+	/datum/element/weather_listener
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/admin/verbs/possess.dm:26]: I might have accidentally left this in when code-diving

File: code/datums/atmosphere/_atmosphere.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Floofies) [reply code/datums/atmosphere/_atmosphere.dm:60]: This whole section is wack. I'll review this later

**Review by Floofies — DISMISSED** [review]

File: code/datums/atmosphere/_atmosphere.dm [hunk 5]
```diff
@@ -50,12 +49,17 @@
 		amount = CEILING(amount, 0.1)
 
 		ASSERT_GAS(gastype, gasmix)
+		if(!gaslist[gastype])
+			gaslist[gastype] = list()
+			gaslist[gastype][MOLES] = 0
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/datums/atmosphere/_atmosphere.dm:54]: Tested with planetary turfs, the two suggestions about this section of code work fine.

File: code/datums/atmosphere/_atmosphere.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/datums/atmosphere/_atmosphere.dm]: Yea, ASSERT_GAS is bugged.

**Comment by nevimer** [comment]
merge conflicts, I think we can merge this ASAP

**Comment by RimiNosha** [comment]
Next up: More maps and ruins, and just content in general. This PR was already huge enough, so I decided to forego as much mapped content as possible. to prevent this killing the review requester.

**Pull request merged**